### ISSUE TITLE
[Feature request] Add a setting to disable Spotify API completely

### ISSUE THREAD

**Issue by ginqus** [Vexcited/better-spotify-genres#10]
I'm from a country that's geoblocked by Spotify. Because of that, Spotify API usage is impossible (without a VPN), and the genres don't show at all.

The main issue I'm facing is the fact that the extension, even if using LastFM as a genre provider, uses Spotify API to create hyperlinks that point to The Sounds Of Spotify playlists for the genres (or at least I assume it does from the code). Because of this, the genres cannot be shown.

So, it would be useful to have an option to disable the hyperlinks, or maybe display the genres without hyperlinks if api.spotify.com search fails.

I'm sorry if I made some false assumptions, I know nothing about TypeScript, but I've modified the code ("ChatGPT modified it for me" is more correct though), and the genres now show up (without hyperlinks). [Here's the build](https://drive.google.com/file/d/1NhgXRLX0-08Gpn_hQI7G6sWHgSY4hCI5/view?usp=drivesdk), I'll upload the code later.

Thank you for the extension and your hard work!

**Comment by vexcited** [comment]
Hello ! Thanks for the suggestions, can you maybe push the modifications in a pull request so we can review the code together ?

**Comment by ginqus** [comment]
> Hello ! Thanks for the suggestions, can you maybe push the modifications in a pull request so we can review the code together ?

Well, I simply removed Spotify API from the code altogether, just so that it could work for me, so a pull request is too much. But I can make a fork later once I get home and link it. It's a very dirty workaround and it doesn't allow switching between LastFM and Spotify API.

**Comment by ginqus** [comment]
Here, I've created a fork: https://github.com/ginqus/better-spotify-genres-lastfm

I don't want to make a PR because my fork removes Spotify API completely, the user can't control this. I would love to help you, but unfortunately I have never worked with TypeScript before, so all I can do is kindly ask you to implement what I described previously. There's no hurry of course, I understand that debugging something to work in a country with sanctions, while not being in said country can be frustrating, so it's just a suggestion. It's fine if you decide not to add this, too. Thank you either way :)

**Comment by vexcited** [comment]
Sure! I'm going to let this issue open in case I get some time to work on this, thanks you!

**Comment by FiftyFour7250** [comment]
I support that. But not because of local restrictions, as in your case, but because for some reason I see many more songs with genres on the version you uploaded to Google Drive with the Spotify API removed. At the same time, there are some songs for which genres are no longer displayed, even though they were displayed when the Spotify API was enabled. 

I believe the best solution at the moment is to use two extensions at the same time. I am currently using the version above without Spotify API and another similar extension: https://github.com/LucasOe/spicetify-genres

Thus, what Spotify API cannot find is displayed through the extension version from Google Drive via Last FM API, and what Last FM cannot find is displayed through Spotify API of the second extension. Sometimes this results in genre names being layered on top of each other from two extensions at once, but in most cases it remains readable.

But, of course, it would be better if someone made a good extension that would take genres from several sources independently of each other...

**Comment by vexcited** [comment]
I understand! I'm looking towards having a switch for each provider so they all get merged when displaying the genres. Last FM genres could be more accurate but there's more chance they won't have a "The sound of ..." playlist made by Spotify, so that's a bit sad but whatever

**Comment by ginqus** [comment]
Yes, that would be amazing! I've now found a free DNS that works as a proxy for Spotify API so now the extension works for me and I've seen just how inaccurate Spotify genres are. The ability to use Last.FM when available and Spotify as a fallback would be very useful

**Comment by vexcited** [comment]
As said in https://github.com/Vexcited/better-spotify-genres/issues/21#issuecomment-3761795911, we'll have no choice than disable the Spotify API entirely in this extension to keep it working.
